pkg/validation/spec.go

```diff
@@ -0,0 +1,22 @@
+package validation
+
+import (
+	"context"
+
+	"github.com/giantswarm/apiextensions/v3/pkg/apis/application/v1alpha1"
+)
+
+type Interface interface {
+	// ValidateApp checks if given App CR is valid. It returns nil if it is.
+	//
+	// It returns error matched by IsValidation if the App CR is invalid.
+	//
+	// It returns error matched by IsAppConfigMapNotFound if the ConfigMap
+	// containing values to template is not found in the cluster. Usually
+	// it should be generated by app-operator after a short delay.
```

Comment:
From what I know, app-operator doesn't create configmaps belong to the app CRs (e.g. cluster configmap) . there are t other components but in most cases it's cluster-operator.

```suggestion
	// it should be generated by cluster-operator after a short delay.
```

Reply:
Yes the cluster values configmap and kubeconfig secret are generated by cluster-operator not app-operator.